core/sched_round_robin.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2020 TUBA Freiberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys
+ * @{
+ *
+ * @file
+ * @brief       Round Robin Scheduler implementation
+ *
+ * @author      Karl Fessel <karl.fessel@ovgu.de>
+ *
+ * @}
+ */
+
+#include "thread.h"
+#include "xtimer.h"
+#include "clist.h"
+#include "sched_round_robin.h"
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+static void sched_round_robin_cb(void * d);
+
+static xtimer_t rr_timer = {.callback = sched_round_robin_cb};
+
+static uint8_t current_rr_priority = 0xff;
+
+static void sched_round_robin_cb(void * d)
+{
+    (void)d;
+    /*
+     * reorder current Round Robin priority
+     * (put the current thread at the end of the run queue of its priority)
+     * and setup the schedueler to schedule when returning from the IRQ
+     */
+    if (current_rr_priority != 0xff) {
+        DEBUG_PUTS("Round_Robin_cb");
+        clist_lpoprpush(&sched_runqueues[current_rr_priority]);
+        thread_t * active_thread = thread_get_active();
```

Comment:
```suggestion
        thread_t *active_thread = thread_get_active();
```

Reply:
i still not understand why you like to put the type close to the name  and have no space between type and name